### PR TITLE
Semgrep

### PR DESCRIPTION
**Pull request by pzhou-splunk**
### Notes
- A common pattern is to read and write file paths obtained from the
platform's [vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API. There's around 100 plus apps that do something like

    ```
    def get_file_path():
        files = phantom.get_vault_info(...)
        for f in files:
           if CONDITION:
              return = f['file_path']
     ...
    def handle_action(): 
        fp = get_file_path()
        ...
        with open(fp) as f:
            ...
   ```
  and Semgrep isn't able to detect the call to the `phantom.get_vault_info(...)` from within the call stack. Since there's too many false positives with the current `filesystem-access` check, and I haven't figured out a way to support the above pattern, I'm removing the rule for now.

- Adding a rule to check for https://splunk.atlassian.net/browse/PAPP-24779
- Updating logging-sensitive-data rule to not consider string literals

### CODE REVIEW

File: semgrep/rules/python/filesystem/path_traversal.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    payload = json.loads(request.POST.get('payload'))
+
+    # ruleid: path-traversal
+    with open(payload) as f:
+        f.write('foo')
+
+
+def handle_request_2(request):
+    payload = json.loads(request.POST.get('payload'))
+    a = payload.get('a')
```
Comment (rgil-splunk) [inline semgrep/rules/python/filesystem/path_traversal.py:17]: would this rule also flag `payload['a']` (the other common form of dictionary access)?

**Review by rgil-splunk — APPROVED** [review]
> Since there's too many false positives with the current filesystem-access check, and I haven't figured out a way to support the above pattern, I'm removing the rule for now.

did you reopen a ticket for it?

also, while not ideal what do you think about preemptively adding the semgrep ignore comment to all the existing cases (basically push it to the next branch) and letting people add the exception when they introduce new instances of it? I ask since the alternative would be manually having to check this, no?

**Comment by pzhou-splunk** [comment]
> did you reopen a ticket for it?
I realized it would be simpler if we just ran these rules through the github pipeline, then we don't need to open a separate MR with ProdSec and wait for them to update their Docker image

>also, while not ideal what do you think about preemptively adding the semgrep ignore comment to all the existing cases (basically push it to the next branch) and letting people add the exception when they introduce new instances of it? I ask since the alternative would be manually having to check this, no?

The issue would be manually combing through the findings to distinguish between false/true positive 😅 We can add a related question to the PR template for now. Ideally, filesystem access should be control by the run-time environment and not statically 😦

File: semgrep/rules/python/filesystem/path_traversal.py (hunk 1 above)
Comment (pzhou-splunk) [reply semgrep/rules/python/filesystem/path_traversal.py:17]: yep

**Comment by rgil-splunk** [comment]
> Ideally, filesystem access should be control by the run-time environment and not statically

that's a good point. especially now that we've had a push to be as hands off as possible in the developer supported community apps. kind of raises a more general platform architecture question of what degree of sandboxing we have. it's not the end of the world if an app nukes itself or its own state but it really shouldn't be able to mess with other apps or the rest of the system's code

**Pull request merged**